pkg/scheduler/scheduler_test.go

```diff
@@ -9345,3 +9348,297 @@ func TestScheduleForTASCohorts(t *testing.T) {
+			},
+		},
+		"multiple workloads multiple queues": {
+			workloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("wl1", "default").
+					Queue("lq-a").
+					PodSets(*utiltesting.MakePodSet("one", 1).
+						Request(corev1.ResourceCPU, "2").
+						Obj()).
+					Obj(),
+				*utiltesting.MakeWorkload("wl2", "default").
+					Queue("lq-b").
+					PodSets(*utiltesting.MakePodSet("one", 1).
+						Request(corev1.ResourceCPU, "3").
+						Obj()).
+					Obj(),
+				*utiltesting.MakeWorkload("wl3", "default").
+					Queue("lq-a").
+					PodSets(*utiltesting.MakePodSet("one", 1).
+						Request(corev1.ResourceCPU, "1").
+						Request(corev1.ResourceMemory, "512Mi").
+						Obj()).
+					Obj(),
+				*utiltesting.MakeWorkload("wl4", "default").
+					Queue("lq-b").
+					PodSets(*utiltesting.MakePodSet("one", 1).
+						Request(corev1.ResourceCPU, "1").
+						Obj()).
+					Obj(),
+			},
+			wantScheduled: []workload.Reference{"default/wl1", "default/wl2", "default/wl3", "default/wl4"},
+			wantAssignments: map[workload.Reference]kueue.Admission{
+				"default/wl1": *utiltesting.MakeAdmission("cq1", "one").Assignment(corev1.ResourceCPU, "default", "2").Obj(),
+				"default/wl2": *utiltesting.MakeAdmission("cq1", "one").Assignment(corev1.ResourceCPU, "default", "3").Obj(),
+				"default/wl3": *utiltesting.MakeAdmission("cq1", "one").Assignment(corev1.ResourceCPU, "default", "1").Assignment(corev1.ResourceMemory, "default", "512Mi").Obj(),
+				"default/wl4": *utiltesting.MakeAdmission("cq1", "one").Assignment(corev1.ResourceCPU, "default", "1").Obj(),
+			},
+			wantPenalties: map[string]corev1.ResourceList{
+				"default/lq-a": calculateExpectedPenalty([]corev1.ResourceList{
+					{corev1.ResourceCPU: resource.MustParse("2")},
+					{
+						corev1.ResourceCPU:    resource.MustParse("1"),
+						corev1.ResourceMemory: resource.MustParse("512Mi"),
+					},
+				}, afsConfig),
+				"default/lq-b": calculateExpectedPenalty([]corev1.ResourceList{
+					{corev1.ResourceCPU: resource.MustParse("3")},
+					{corev1.ResourceCPU: resource.MustParse("1")},
+				}, afsConfig),
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			features.SetFeatureGateDuringTest(t, features.AdmissionFairSharing, true)
+
+			clientBuilder := utiltesting.NewClientBuilder().
+				WithLists(&kueue.WorkloadList{Items: tc.workloads}, &kueue.ClusterQueueList{Items: clusterQueues}, &kueue.LocalQueueList{Items: queues}).
+				WithObjects(
+					utiltesting.MakeNamespace("default"),
+				)
+			cl := clientBuilder.Build()
+			cqCache := cache.New(cl)
+			qManager := queue.NewManager(cl, cqCache)
+
+			ctx, log := utiltesting.ContextWithLog(t)
+			for _, q := range queues {
+				if err := qManager.AddLocalQueue(ctx, &q); err != nil {
+					t.Fatalf("Inserting queue %s/%s in manager: %v", q.Namespace, q.Name, err)
+				}
+			}
+			for i := range resourceFlavors {
+				cqCache.AddOrUpdateResourceFlavor(log, resourceFlavors[i])
+			}
```

Comment:
```suggestion
			for _, rf := range resourceFlavors {
				cqCache.AddOrUpdateResourceFlavor(log, rf)
			}
```

Just for consistency

Reply:
Done.